include/edm4hep/GenToolInfo.h

```diff
@@ -0,0 +1,48 @@
+#ifndef EDM4HEP_GENTOOLINFO_H
+#define EDM4HEP_GENTOOLINFO_H
+
+#include "edm4hep/Constants.h"
+#include "podio/Frame.h"
+#include <string>
+#include <vector>
+
+namespace edm4hep {
+
+struct GenToolInfo {
+  std::string name;
+  std::string version;
+  std::string description;
+
+  GenToolInfo(){};
+  GenToolInfo(const std::string& name, const std::string& version, const std::string& description) :
+      name(name), version(version), description(description){};
+};
+
+const std::vector<GenToolInfo> getGenToolInfos(const podio::Frame& frame) {
+  auto toolInfos = std::vector<GenToolInfo>();
+  const auto names = frame.getParameter<std::vector<std::string>>(generatorToolNameLabel);
+  const auto versions = frame.getParameter<std::vector<std::string>>(generatorToolVersionLabel);
+  const auto descriptions = frame.getParameter<std::vector<std::string>>(generatorToolDescriptionLabel);
+  for (unsigned int i = 0; i < names.size(); ++i) {
+    toolInfos.emplace_back(names[i], versions[i], descriptions[i]);
```

Comment:
This will need to take into account the changes that have happened in AIDASoft/podio#580 as `getParameter` now returns a `std::optional`